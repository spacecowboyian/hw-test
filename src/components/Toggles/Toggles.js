import React from 'react'
import { Row, Col } from 'react-flexbox-grid';
import { css } from 'emotion/macro';
import { FiSliders } from "react-icons/fi";

import { spacing } from '../../Styles';

//TODO: snapshots prop should accept snapshots array

const iconSliders = css`
  font-size: 2.2em;
  color: #00E0C2;
`

const toggles = css`
  list-style: none;
  padding: 0;
  margin: 0;
  display: flex;
`

const toggle = css`
  color: #A6AEAE;
  border: 1px solid #979797;
  padding: calc(${spacing}*3) calc(${spacing}*5);
  margin-right: calc(${spacing}*2);
  border-radius: 6px;
  font-size: 13px;
  font-weight: 500;
  white-space: nowrap;
`
const toggleActive = css`
  ${toggle}
  border-color: #00E0C2;
  background-color: #00E0C2;
  color: #007371;
`

const toggleSlider = css`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`

export default function Toggles() {
  return (

    <Row>
      <Col xs={12} sm={11}>
        <ul className={toggles}>
          <li className={toggleActive}>Premium Range</li>
          <li className={toggle}>Markets</li>
          <li className={toggleActive}>Industries</li>
          <li className={toggle}>Products</li>
        </ul>
      </Col>
      <Col className={toggleSlider} xs={12} sm={1}>
        <FiSliders className={iconSliders} />
      </Col>
    </Row>

  )
}
