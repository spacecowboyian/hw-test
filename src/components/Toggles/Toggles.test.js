import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Toggles from './Toggles';

afterEach(cleanup);


it('renders correctly', () => {
  const tree = renderer
    .create(<Toggles />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
