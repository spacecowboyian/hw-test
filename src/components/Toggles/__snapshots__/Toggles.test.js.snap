// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12 col-sm-11"
  >
    <ul
      className="css-ryds10-toggles"
    >
      <li
        className="css-1os4whm-toggle-toggleActive"
      >
        Premium Range
      </li>
      <li
        className="css-1n670sg-toggle"
      >
        Markets
      </li>
      <li
        className="css-1os4whm-toggle-toggleActive"
      >
        Industries
      </li>
      <li
        className="css-1n670sg-toggle"
      >
        Products
      </li>
    </ul>
  </div>
  <div
    className="col-xs-12 col-sm-1 css-kbtrig-toggleSlider"
  >
    <svg
      className="css-8m7lne-iconSliders"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="4"
        x2="4"
        y1="21"
        y2="14"
      />
      <line
        x1="4"
        x2="4"
        y1="10"
        y2="3"
      />
      <line
        x1="12"
        x2="12"
        y1="21"
        y2="12"
      />
      <line
        x1="12"
        x2="12"
        y1="8"
        y2="3"
      />
      <line
        x1="20"
        x2="20"
        y1="21"
        y2="16"
      />
      <line
        x1="20"
        x2="20"
        y1="12"
        y2="3"
      />
      <line
        x1="1"
        x2="7"
        y1="14"
        y2="14"
      />
      <line
        x1="9"
        x2="15"
        y1="8"
        y2="8"
      />
      <line
        x1="17"
        x2="23"
        y1="16"
        y2="16"
      />
    </svg>
  </div>
</div>
`;
